### ISSUE TITLE
Parse error for multiline object props on React components

### ISSUE THREAD

**Issue by mike-fam** [mdx-js/eslint-mdx#435]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

2.0.5

### Link to runnable example

https://github.com/mike-fam/eslint-mdx-multiline-prop-error

### Steps to reproduce

1. Create a new workspace with npm and install `eslint` with `eslint-plugin-mdx`
2. Eslint configuration:
 ```
{
	"overrides": [
		{
			"files": ["*.mdx"],
			"extends": "plugin:mdx/recommended"
		}
	],
	"rules": {}
}
 ```
3. Create an mdx file with a react component with a multiline object prop e.g.
 ```
<button
	style={{
		display: 'block'
	}}
>
	Test
</button>
 ```
4. Lint the file, note the error: `Parsing error: Expected value to be truthy`

### Expected behavior

The file should pass linting without any problems, as this is valid mdx syntax.

### Actual behavior

Linting produces the following error:
`0:0  error  Parsing error: Expected value to be truthy`

### Runtime

Node v16, Node v14

### Package manager

yarn v1

### OS

macOS

### Build and bundle tools

webpack

**Comment by JounQin** [comment]
Thanks for reporting issue by using the template, I'm on a vocation these days, I'll check later soon.

**Comment by JounQin** [comment]
It seems related to `tab`, if I change the source codes with spaces there will be no issue:

```jsx
<button
  style={{
    display: "block",
  }}
>
  Test
</button>
```

The assertion error is thrown at https://github.com/mdx-js/eslint-mdx/blob/a1998c787f0049ae0cd705fc51f079c63ded75b2/packages/eslint-mdx/src/worker.ts#L491 where `attrValEnd` comes from `data.estree.range` directly, so could it be an issue from `micromark-extension-mdx-expression`? @wooorm

**Comment by mike-fam** [comment]
Hi @JounQin, thanks for all your great work and investigating this. Is there any new update on this? For now, do you recommend replacing all the tabs in our stories with spaces or waiting for a new fix to be released?

**Comment by morganfeeney** [comment]
Hey @mike-fam did you manage to work around this one? I'm facing the same thing.

**Comment by Ayc0** [comment]
I investigated a bit this issue.

With the following code:
```mdx
-   list

    <Components data={`
    `} />
```

I also have the error `Parsing error: Expected value to be truthy`

This crashes because of this line: https://github.com/mdx-js/eslint-mdx/blob/9331895e902e4eb08cf2343bd7ccac2adf83af40/packages/eslint-mdx/src/worker.ts#L413.

When adding console.logs before and after `attrValStart = prevCharOffset(attrValStart - 1); attrValEnd = nextCharOffset(attrValEnd);`, here is what I have:

<img width="458" alt="image" src="https://user-images.githubusercontent.com/22725671/212447550-e137bbf6-f011-40fe-9573-b6c263cbc7a0.png">

I think that the reason of the offset is because when writing ``<Component data={`\n    `} />``, `acorn` returns this node:
<img width="349" alt="image" src="https://user-images.githubusercontent.com/22725671/212447711-8cf9b62a-ce3d-4916-887f-986848a7ccec.png">

But the offsets start & end on the `data` attribute (32, 35) don't match the right start & end (end should have been 32 38):
<img width="440" alt="image" src="https://user-images.githubusercontent.com/22725671/212447755-914c6eaf-0113-4016-a434-10e8480f7a86.png">

I'm not sure why the spaces are dropped in this estree (which comes from this processor https://github.com/mdx-js/eslint-mdx/blob/9331895e902e4eb08cf2343bd7ccac2adf83af40/packages/eslint-mdx/src/worker.ts#L79).

**Comment by mike-fam** [comment]
Hi @JounQin @wooorm sorry for bothering you. Is there any update on this issue?

**Comment by remcohaszing** [comment]
If I’m correct, this repo is mainly maintained by @JounQin. However, we haven’t heard from him in a while.

Maybe we should create and pin a _Looking for maintainers_ issue?

**Comment by JounQin** [comment]
As I and @Ayc0 discovered above, it is not an internal issue but due to acorn return incorrect location info, but I'm not sure if it comes from `acorn` itself or `micromark-extension-mdx-expression` instead because the location info is restored at

https://github.com/micromark/micromark-extension-mdx-expression/blob/88c175f9ac684efc9306e23642807657df71d6a6/packages/micromark-util-events-to-acorn/dev/index.js#L216

cc @wooorm

**Comment by mike-fam** [comment]
@JounQin @wooorm Would you guys mind creating an issue in either `acorn` or `micromark-extension-mdx-expression`, since I know little context to how the eslint-mdx plugin is implemented and how it uses those dependencies

**Comment by wooorm** [comment]
I dunno where this problem is, I’d appreciate a reduced test case. If it isn’t in `eslint-mdx`, I’d appreciate a test case that doesn’t use `eslint-mdx`?

**Comment by JounQin** [comment]
Working on a minimal reproduction without `eslint-mdx`.

**Comment by JounQin** [comment]
@wooorm Please check https://github.com/JounQin/test/blob/mdx-tab/test.mjs#L29-L32

Besides, https://github.com/mdx-js/eslint-mdx/issues/435#issuecomment-1382644367 could be another range issue not related to `tab` but `list`.

**Comment by wooorm** [comment]
@JounQin did you see my above commit?

**Comment by JounQin** [comment]
@wooorm Thanks! It seems working to me now. @mike-fam @morganfeeney @Ayc0 Can you help to confirm? You may need to delete your `npm-lock.json` or `yarn.lock`, etc to upgrade all related packages to confirm.
